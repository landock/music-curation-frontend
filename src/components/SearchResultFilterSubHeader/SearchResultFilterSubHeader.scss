//SearchResultFilterSubHeader
.ui.text.vertical.menu a {
    color: white;
}