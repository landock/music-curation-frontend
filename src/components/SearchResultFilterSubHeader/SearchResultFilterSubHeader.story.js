import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import SearchResultFilterSubHeader from '.';
import { tags } from '../../fixtures/test_data';

storiesOf('SearchResultFilterSubHeader', module).add('default', () => (
  <SearchResultFilterSubHeader tags={tags} />
));
