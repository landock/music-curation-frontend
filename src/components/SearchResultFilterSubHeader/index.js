import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Grid, Menu, Form, Header } from 'semantic-ui-react';
import TagCollection from '../TagCollection';

import './SearchResultFilterSubHeader.css';

class SearchResultFilterSubHeader extends PureComponent {
  render() {
    return (
      <Grid columns="4">
        <Grid.Row color="black">
          <Grid.Column width={2}>
            <Menu
              inverted
              color="black"
              vertical
              text
              items={[
                { name: 'All', key: 'all' },
                { name: 'Tracks', key: 'tracks' },
                { name: 'Artist', key: 'artist' },
                { name: 'Albums', key: 'albums' },
                { name: 'Label', key: 'label' },
              ]}
            />
          </Grid.Column>
          <Grid.Column width={this.props.tags ? 4 : 7}>
            <Form inverted widths="equal">
              <Form.Input label="Dates" type="range" />
              <Form.Group>
                <Form.Select
                  label="Country"
                  options={[{ key: 'af', value: 'af', text: 'Afghanistan' }]}
                  fluid
                />
                <Form.Select
                  label="City"
                  options={[
                    { key: 'kandahar', value: 'kandahar', text: 'Kandahar' },
                  ]}
                  fluid
                />
              </Form.Group>
            </Form>
          </Grid.Column>
          <Grid.Column width={5}>
            <Header inverted size="small">
              Release type
            </Header>
            <Grid columns="equal">
              <Grid.Row>
                <Grid.Column>
                  <Menu
                    inverted
                    vertical
                    text
                    items={[
                      { name: 'Karaoke', key: 'karaoke' },
                      { name: 'Soundtrack', key: 'soundtrack' },
                      { name: 'Compilation', key: 'compilation' },
                      { name: 'Single', key: 'single' },
                      { name: 'B-Side', key: 'b-side' },
                    ]}
                  />
                </Grid.Column>
                <Grid.Column>
                  <Menu
                    inverted
                    vertical
                    text
                    items={[
                      { name: 'Collaboration', key: 'collaboration' },
                      { name: 'Related', key: 'related' },
                      { name: 'Remix', key: 'remix' },
                      { name: 'Live', key: 'live' },
                      { name: 'Acoustic', key: 'acoustic' },
                    ]}
                  />
                </Grid.Column>
              </Grid.Row>
            </Grid>
          </Grid.Column>
          <Grid.Column width={this.props.tags ? 5 : 2}>
            <TagCollection tags={this.props.tags} />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}

SearchResultFilterSubHeader.propTypes = { tags: PropTypes.array };

// SearchResultFilterSubHeader.defaultProps = {
//   tags,
// };
export default SearchResultFilterSubHeader;
