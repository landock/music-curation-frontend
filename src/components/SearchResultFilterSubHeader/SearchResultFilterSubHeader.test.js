import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchResultFilterSubHeader from '.';

it('SearchResultFilterSubHeader: default', () => {
  const component = renderer.create(<SearchResultFilterSubHeader />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
