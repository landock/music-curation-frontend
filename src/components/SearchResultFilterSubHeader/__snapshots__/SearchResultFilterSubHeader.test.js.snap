// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultFilterSubHeader: default 1`] = `
<div
  className="ui left aligned four column grid"
>
  <div
    className="black row"
  >
    <div
      className="two wide column"
    >
      <div
        className="ui black inverted text vertical menu"
      >
        <a
          className="item"
          onClick={[Function]}
        >
          All
        </a>
        <a
          className="item"
          onClick={[Function]}
        >
          Tracks
        </a>
        <a
          className="item"
          onClick={[Function]}
        >
          Artist
        </a>
        <a
          className="item"
          onClick={[Function]}
        >
          Albums
        </a>
        <a
          className="item"
          onClick={[Function]}
        >
          Label
        </a>
      </div>
    </div>
    <div
      className="seven wide column"
    >
      <form
        action={undefined}
        className="ui inverted equal width form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Dates
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              disabled={undefined}
              onChange={[Function]}
              required={undefined}
              tabIndex={undefined}
              type="range"
            />
          </div>
        </div>
        <div
          className="fields"
        >
          <div
            className="field"
          >
            <label
              htmlFor={undefined}
            >
              Country
            </label>
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              aria-multiselectable={undefined}
              className="ui fluid selection dropdown"
              content={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={undefined}
              role="listbox"
              tabIndex={0}
              type={undefined}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              />
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Afghanistan
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <label
              htmlFor={undefined}
            >
              City
            </label>
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              aria-multiselectable={undefined}
              className="ui fluid selection dropdown"
              content={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={undefined}
              role="listbox"
              tabIndex={0}
              type={undefined}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              />
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Kandahar
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="five wide column"
    >
      <div
        className="ui small inverted header"
      >
        Release type
      </div>
      <div
        className="ui equal width grid"
      >
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="ui inverted text vertical menu"
            >
              <a
                className="item"
                onClick={[Function]}
              >
                Karaoke
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                Soundtrack
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                Compilation
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                Single
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                B Side
              </a>
            </div>
          </div>
          <div
            className="column"
          >
            <div
              className="ui inverted text vertical menu"
            >
              <a
                className="item"
                onClick={[Function]}
              >
                Collaboration
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                Related
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                Remix
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                Live
              </a>
              <a
                className="item"
                onClick={[Function]}
              >
                Acoustic
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="two wide column"
    >
      <span>
        No tags
      </span>
    </div>
  </div>
</div>
`;
